xls/contrib/mlir/IR/xls_ops.cc

```diff
@@ -315,6 +315,16 @@ LogicalResult ArrayIndexOp::canonicalize(ArrayIndexOp op,
   return LogicalResult::failure();
 }
 
+LogicalResult TupleIndexOp::canonicalize(TupleIndexOp op,
+                                         PatternRewriter& rewriter) {
+  // tuple_index(tuple($x1, $x2, ...), index=N) = $xN
+  if (auto defining_op = op.getOperand().getDefiningOp<TupleOp>()) {
+    rewriter.replaceAllOpUsesWith(op, defining_op->getOperand(op.getIndex()));
+    return LogicalResult::success();
```

Comment:
nit: just use success(). it's a free function in the mlir:: namespace (that we are in here).